virtualizarr/readers/zarr.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import annotations
+
+import asyncio
+from pathlib import Path  # noqa
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Hashable,
+    Iterable,
+    Mapping,
+    Optional,
+)
+
+import numpy as np
+from xarray import Dataset, Index
+from zarr.api.asynchronous import open_group as open_group_async
+from zarr.core.metadata import ArrayV3Metadata
+
+from virtualizarr.manifests import (
+    ChunkManifest,
+    ManifestArray,
+    ManifestGroup,
+    ManifestStore,
+)
+from virtualizarr.manifests.manifest import validate_and_normalize_path_to_uri  # noqa
+from virtualizarr.readers.api import VirtualBackend
+from virtualizarr.vendor.zarr.core.common import _concurrent_map
+
+if TYPE_CHECKING:
+    pass
+
+FillValueT = bool | str | float | int | list | None
+
+ZARR_DEFAULT_FILL_VALUE: dict[str, FillValueT] = {
+    # numpy dtypes's hierarchy lets us avoid checking for all the widths
+    # https://numpy.org/doc/stable/reference/arrays.scalars.html
+    np.dtype("bool").kind: False,
+    np.dtype("int").kind: 0,
+    np.dtype("float").kind: 0.0,
+    np.dtype("complex").kind: [0.0, 0.0],
+    np.dtype("datetime64").kind: 0,
+}
+
+
+if TYPE_CHECKING:
+    import zarr
+
+
+async def get_chunk_mapping_prefix(zarr_array: zarr.AsyncArray, filepath: str) -> dict:
+    """Create a dictionary to pass into ChunkManifest __init__"""
+
+    # TODO: For when we want to support reading V2 we should parse the /c/ and "/" between chunks
+
+    prefix = zarr_array.name.lstrip("/") + "/c/"
```

Comment:
I think this will fail for scalars. I recently learned that zarr v3 uses just the path `c` to hold scalars (because `c/0` would imply that there is at least one dimension to have a chunk index along).

Reply:
ISTG it's in the zarr spec but I can't find it now

Reply:
It got moved from the core spec to the chunk_key_encoding extension - https://zarr-specs.readthedocs.io/en/latest/v3/chunk-key-encodings/default/index.html

Reply:
I started a test to try to check this. It's xfailed for now. Can we infer a scalar `c` chunk from store.info or?

Reply:
Added a test for this: https://github.com/zarr-developers/VirtualiZarr/blob/ccc92796bc1f207c47a92660e8dafa3c8779ed9b/virtualizarr/tests/test_readers/test_zarr.py#L79

Reply:
1. That's what I ended up using!
2. Yup!

Reply:
it's a little verbose but totally clear. LGTM!